services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/tormaystarkastelu/TormaystarkasteluPaikkaServiceImpl.kt

```diff
@@ -25,47 +24,160 @@ class TormaystarkasteluPaikkaServiceImpl : TormaystarkasteluPaikkaService {
         return luokitteluTulosComplete
     }
 
+    internal fun getKatuluokkaLuokitteluTulos(hanke: Hanke): List<Luokittelutulos> {
```

Comment:
Tämä on erittäin pitkä funktio, joka sisältää monia logiikkahaaroja. Voisiko olla mahdollista pilkkoa vielä pienempiin osiin tai siirtää logiikkaa enumeihin tms?

Lisäksi, tässä on useita return-lauseita. Olisi mielestäni parempi, jos olisi vain yksi return. 

Lisäksi, funktio palauttaa listan, mutta nähdäkseni siinä on aina vain max. 1 item sisässä. Olisiko siis parempi palauttaa vain yksi Luokittelutulos?

Reply:
Korjattu, että palauttaa vain yhtä otusta. Samalla joutui alkaa käsitellä virhetilannetta, jos ei katuluokkaa saadakaan kasaan.